### PR TITLE
audit: disallow capitalization and period in conflicts_with reason

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Check the grammar of `conflicts_with` reasons, so that

`conflicts_with "gdal", :because => "both install a cpl_conv.h header"`

is printed by `brew info <formula>` as

```
Conflicts with:
  gdal (because both install a cpl_conv.h header)
```

### CODE REVIEW

**Comment by gromgit** [comment]
I think "because" can be dropped. The parentheses already set the explanation apart.

File: Library/Homebrew/rubocops/conflicts.rb [hunk 1]
```diff
@@ -15,11 +15,25 @@ class Conflicts < FormulaCop
...
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          conflicts_with_node = find_node_method_by_name(body_node, :conflicts_with)
+          return unless conflicts_with_node
+
+          if parameters(conflicts_with_node).last.respond_to? :values
+            reason = string_content(parameters(conflicts_with_node).last.values.first)
+            first_word = reason.split.first
+
+            if reason.match?(/\A[A-Z]/)
+              problem "'#{first_word}' from the conflicts_with reason should be '#{first_word.downcase}'."
```
Comment (Rylan12) [inline Library/Homebrew/rubocops/conflicts.rb:27]: Very minor but you might want to surround `conflicts_with` with backticks.

This formats it differently then the normal text and, in my opinion, makes it easier to read.

```suggestion
              problem "'#{first_word}' from the `conflicts_with` reason should be '#{first_word.downcase}'."
```

**Comment by EricFromCanada** [comment]
At the moment it only examines the first `conflicts_with`, so I'm also looking at that. (This is my first foray into audit-writing.)

**Comment by Rylan12** [comment]
> At the moment it only examines the first `conflicts_with`, so I'm also looking at that. (This is my first foray into audit-writing.)

```find_every_method_call_by_name``` maybe?

https://github.com/Homebrew/brew/blob/59f622d4baf0b417ccba5b69f4aa3bfeddde902e/Library/Homebrew/rubocops/extend/formula.rb#L110-L119

File: Library/Homebrew/rubocops/conflicts.rb (hunk 1 above)
Comment (EricFromCanada) [reply Library/Homebrew/rubocops/conflicts.rb:27]: This was lifted directly from `keg_only.rb`, so someone might want to apply this change there also.

Comment (Rylan12) [reply Library/Homebrew/rubocops/conflicts.rb]: Completely agree. Thanks for pointing that out!

Comment (Rylan12) [reply Library/Homebrew/rubocops/conflicts.rb]: It looks like the changes I suggested here could also be added to `keg_only.rb`, so I'll work on a PR to implement those there.

**Comment by EricFromCanada** [comment]
Remind me why the Ubuntu VM doesn't appear to be fetching the latest homebrew/core…?

**Comment by Rylan12** [comment]
> Remind me why the Ubuntu VM doesn't appear to be fetching the latest homebrew/core…?

`homebrew/core` on Linux is actually pointing to the [`linuxbrew/core`](https://github.com/Homebrew/linuxbrew-core) repository, so it won't get formula changes until they are made in the `linuxbrew/core` repo.

`homebrew/core` is manually merged into `linuxbrew/core` every so often (usually at least once a day), so I think you just need to wait until the changes in the `homebrew/core` repo are merged in (or merge them in yourself, maybe).

There's some more information about the process [here](https://docs.brew.sh/Homebrew-linuxbrew-core-Maintainer-Guide).

**Comment by EricFromCanada** [comment]
Linuxbrew changes suggested in Homebrew/linuxbrew-core#20673 and Homebrew/linuxbrew-core#20674.

**Comment by EricFromCanada** [comment]
Auto-correct is implemented and tests are passing.

**Comment by MikeMcQuaid** [comment]
Great work, thanks @EricFromCanada!

**Pull request merged**